Tests/Common/Util/RingBufferTest.swift

```diff
@@ -0,0 +1,59 @@
+@testable import CioInternalCommon
+import Foundation
+import SharedTests
+import XCTest
+
+class RingBufferTest: UnitTest {
+    func test_EnqueueAndDequeue_ShouldHandleElementsInFIFOOrder() {
+        var ringBuffer = RingBuffer<Int>(capacity: 3)
+        ringBuffer.enqueue(1)
+        ringBuffer.enqueue(2)
+
+        XCTAssertEqual(ringBuffer.dequeue(), 1, "Dequeued element should be 1")
+        XCTAssertEqual(ringBuffer.dequeue(), 2, "Dequeued element should be 2")
+        XCTAssertNil(ringBuffer.dequeue(), "Buffer should be empty")
+    }
+
+    func test_EnqueueBeyondCapacity_ShouldPushOldestElement() {
```

Comment:
```suggestion
    func test_EnqueueBeyondCapacity_ShouldRemoveOldestElement() {
```
The method name is messing with my brain. The name provides an impression that it should "push the oldest element" which makes me think that the oldest element should be pushed out as output when dequeued but the method removes the first element and pops out in FIFO order when dequeued. 🤔